TLM/TLM/Manager/Impl/LaneConnectionManagerData/LaneEnd.cs

```diff
@@ -0,0 +1,24 @@
+namespace TrafficManager.Manager.Impl.LaneConnectionManagerData {
+    using TrafficManager.Util.Extensions;
+
+    internal struct LaneEnd {
+        internal uint LaneId;
+        internal bool StartNode;
```

Comment:
I would add `readonly` for fields and struct itself, it will ensure that no one come up with idea of modifying keys - dictionary keys should be immutable.